app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java

```diff
@@ -0,0 +1,52 @@
+package swati4star.createpdf.util;
+
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+public class PermissionsUtils {
+    private static final String TAG = "PermissionsUtils";
+    private static Activity mActivity;
```

Comment:
@sidhuparas Please tell why will they leak memory?

Reply:
It’s very common in Android that when you rotate the device or due to any type of configuration changes, the activity instance is destroyed and a new instance will be created. But mActivity would still be holding up the old instance. That would cause Garbage Collector to not kill that instance because it’s still held. This is memory leak. Also read the Android Studio’s warning when you make an Activity instance as static.